airgun/entities/repository.py

```diff
@@ -14,8 +15,22 @@
 
 class RepositoryEntity(BaseEntity):
 
+    def _find_default_http_proxy(self, values):
+        """ Finds the global proxy name and returns updated values accordingly."""
+        proxy = SettingsEntity(self.browser)
+        result = proxy.read(property_name='name = content_default_http_proxy')['table'][0]['Value']
+        global_proxy_name = result.split()[0]
+        if result.split()[0] == 'no':
+            values["repo_content.http_proxy_policy"] = 'Global Default (None)'
+        else:
+            values["repo_content.http_proxy_policy"] = 'Global Default ({})'.format(
+                global_proxy_name)
+        return values
+
     def create(self, product_name, values):
         """Create new repository for product"""
+        if values["repo_content.http_proxy_policy"] == "Global Default":
+            values = self._find_default_http_proxy(values)
```

Comment:
There are a lot of existing tests that don't provide `repo_content.http_proxy_policy`, and all would fail with KeyError exception now.
You need to first check if that key was provided at all:
```python
if values.get("repo_content.http_proxy_policy") == "Global Default":
    values = self._find_default_http_proxy(values)
```

Reply:
@mirekdlugosz updated PR.